Lib/Protocols/IdFTP.pas

```diff
@@ -2771,29 +2771,30 @@ procedure TIdFTP.IssueFEAT;
       Break;
     end;
   end;
-  if FClientInfo.ClientName <> '' then begin
-    if IsExtSupported('CSID') then begin {do not localize}
-
-      SendCmd('CSID ' + FClientInfo.GetCSIDOutput);  {do not localize}
+  if IsExtSupported('CSID') then begin {do not localize}
+      LClnt := FClientInfo.GetCSIDOutput;
+      if LClnt = '' then begin
+        LClnt := 'Name='+gsIdProductName + '; Version=' + gsIdVersion;
+      end;
+      SendCmd('CSID ' + LClnt);  {do not localize}
       if Self.LastCmdResult.NumericCode = 200 then begin
         LBuf := TrimRight(Self.LastCmdResult.Text.Text);
         repeat
            FServerInfo.ServerInfo.Add(TrimLeft(Fetch(LBuf,';')));
         until LBuf = '';
       end;
-    end
-    else
-    begin
+  end
+  else
+  begin
   // send the CLNT command before sending the OPTS UTF8 command.
   // some servers need this in order to work around a bug in
   // Microsoft Internet Explorer's UTF-8 handling
-      if IsExtSupported('CLNT') then begin {do not localize}
-        LClnt := FClientInfo.ClntOutput;
-        if LClnt = '' then begin
-          LClnt := gsIdProductName + ' ' + gsIdVersion;
-        end;
-        SendCmd('CLNT ' + LClnt);  {do not localize}
+    if IsExtSupported('CLNT') then begin {do not localize}
+      LClnt := FClientInfo.ClntOutput;
+      if LClnt = '' then begin
+        LClnt := gsIdProductName + ' ' + gsIdVersion;

```

Comment:
The more I think about it, use of `gsIdProductName` and `gsIdVersion` should probably be moved inside of `GetClntOutput()` as the default return value when `FClientName` is empty.